backend/entityservice/tasks/comparing.py

```diff
@@ -360,49 +370,50 @@ def reindex_using_encoding_ids(recordarray, encoding_id_list):
-            merged_file_iter, merged_file_size \
-                = anonlink.serialization.merge_streams_iter(file_iters, sizes=file_sizes)
-            merged_file_iter = iterable_to_stream(merged_file_iter)
-        elif len(file_iters) == 1:
-            merged_file_iter = file_iters[0]
-            merged_file_size = file_sizes[0]
-        else:
-            return 0, None, None
-
-        result_filename = Config.SIMILARITY_SCORES_FILENAME_FMT.format(
-            generate_code(12))
-        task_span.log_kv({"edges": num_results})
-        log.info("Writing {} intermediate results to file: {}".format(num_results, result_filename))
-
-        mc = connect_to_object_store()
-        try:
-            mc.put_object(
-                Config.MINIO_BUCKET, result_filename, merged_file_iter, merged_file_size)
-        except minio.S3Error as err:
-            log.warning("Failed to store result in minio: {}".format(err))
-            raise
-
-    return num_results, merged_file_size, result_filename
+        # Save results file into minio
+        with new_child_span('save-comparison-results-to-minio'):
+            file_iters = []
+            file_sizes = []
+            for sims, (rec_is0, rec_is1), dp1_ds_idx, dp2_ds_idx in sim_results:
+                num_sims = len(sims)
+
+                if num_sims:
+                    # Make index arrays for serialization
+                    index_1 = array.array('I', (dp1_ds_idx,)) * num_sims
+                    index_2 = array.array('I', (dp2_ds_idx,)) * num_sims
+                    chunk_results = sims, (index_1, index_2), (rec_is0, rec_is1),
+                    bytes_iter, file_size = anonlink.serialization.dump_candidate_pairs_iter(chunk_results)
+                    file_iters.append(iterable_to_stream(bytes_iter))
+                    file_sizes.append(file_size)
+
+            if len(file_iters) > 1:
+                # we need to merge them first into one ordered thingy
+                merged_file_iter, merged_file_size = anonlink.serialization.merge_streams_iter(
+                    file_iters,
+                    sizes=file_sizes)
+                merged_file_iter = iterable_to_stream(merged_file_iter)
+            elif len(file_iters) == 1:
+                merged_file_iter = file_iters[0]
+                merged_file_size = file_sizes[0]
+            else:
+                return 0, None, None
+
+            result_filename = Config.SIMILARITY_SCORES_FILENAME_FMT.format(generate_code(12))
+            task_span.log_kv({"edges": num_results})
+            log.info("Writing {} intermediate results to file: {}".format(num_results, result_filename))
+
+            mc = connect_to_object_store()
+            try:
+                mc.put_object(Config.MINIO_BUCKET, result_filename, merged_file_iter, merged_file_size)
+            except minio.S3Error as err:
+                log.warning("Failed to store result in minio: {}".format(err))
+                raise
```

Comment:
do you think we should wrap that in a separate retry with tenacity? We've come so far already, what if minio is temporarily not available?

Reply:
Yeah that sounds sensible, will do